prettyprinter/src/Prettyprinter/Internal.hs

```diff
@@ -630,6 +630,24 @@ group x = case x of
 -- See https://github.com/quchen/prettyprinter/issues/22 for the  corresponding
 -- ticket.
 
+-- Note [Group: Optimial placement of Union and loss of information]
+-- 
+-- group will create a 'Union' in place or not at all. A 'Union' for the layout
+-- algorithm represents a branch, and if that branch ends up failing, it would
+-- be beneficial to keep it as short as possible. Thus group could make an extra
+-- effort to push the resulting 'Union' further down or eliminate the branch
+-- altogether. The result would be a fewer and smaller branches and thus cheaper
+-- failure for the layout algorithm.
+-- This approach would not increase the cost of a call to group at all as
+-- changesUponFlattening already traverses deep enough already, however, due to
```

Comment:
This isn't very obvious. Wouldn't the new approach increase the "length" of the `Doc` that we need to rebuild on top of the new `Union`?!

Reply:
Ah, I forget the cost of recreating the `Doc`, that whole line is guesswork anyway and should probably be removed. I was only thinking in traversal depth.